### PR TITLE
[Assets] Add additional assets used by qeds package from s3 site

### LINKED ISSUE #173: [environment] Missing Data Files causing Execution Issues
The way the `environment.yml` has been setup seems fragile without many version pinnings. 

I also find it hard to replicate `locally` (i.e. something in this `environment.yml` file is driving `python=3.9` despite `python=3.8` being requested. 

Should probably review to improve stability and package management.

### PR DESCRIPTION
**Pull request by mmcky**
fixes #173 to make assets available via `datascience.quantecon.org` so `qeds` can retrieve them. 

@jlperla I'll wait to see if you approve as this is adding `97Mb` to the git history. 

**Note:** this requires a publish step for the assets to be copied into `gh-pages` host

### CODE REVIEW

**Comment by jlperla** [comment]
All sounds good to me

**Comment by mmcky** [comment]
I am going to close this PR in favour of https://github.com/QuantEcon/qeds/pull/8

**Comment by mmcky** [comment]
@jlperla this PR just blindly migrates the assets from the old site to this site to make them available. 

@Alim-faraji perhaps we should grep for each file in `lectures/_data` (+ thinking through https://github.com/QuantEcon/qeds/pull/8) to remove any assets that may not be used any more.

File: lectures/pandas/groupby.md [hunk 1]
```diff
@@ -292,7 +292,8 @@ Let's apply our new split-apply-combine skills to the airline dataset we
 saw in the {doc}`merge <merge>` lecture.
 
 ```{code-cell} python
-air_dec = qeds.load("airline_performance_dec16")
+url = "https://datascience.quantecon.org/assets/data/air_dec1.csv.zip"
```
Comment (wupeifan) [inline lectures/pandas/groupby.md:295]: why it's `air_dec1`?

File: lectures/pandas/merge.md [hunk 2]
```diff
@@ -561,7 +562,8 @@ We hear you.
 Let's also load a dataset containing information on the actual books.
 
 ```{code-cell} python
-books = qeds.data.load("goodreads_books")
+url = "https://datascience.quantecon.org/assets/data/books.csv"
```
Comment (wupeifan) [inline lectures/pandas/merge.md:565]: Make it `goodread_books`

File: lectures/pandas/merge.md [hunk 3]
```diff
@@ -634,7 +636,8 @@ that describes the cause of all US domestic flight delays
 in November 2016:
 
 ```{code-cell} python
-air_perf = qeds.load("airline_performance_dec16")[["CRSDepTime", "Carrier", "CarrierDelay", "ArrDelay"]]
+url = "https://datascience.quantecon.org/assets/data/air_perf.csv"
+air_perf = pd.read_csv(url)
```
Comment (wupeifan) [inline lectures/pandas/merge.md:640]: The data file is the same as used in the other case. Split this line into 2, one loading data and the other fetching those 4 specific columns

File: lectures/pandas/groupby.md (hunk 1 above)
Comment (wupeifan) [reply lectures/pandas/groupby.md:295]: Could you please keep the same file name as before? thanks

File: lectures/pandas/merge.md [hunk 4]
```diff
@@ -665,7 +668,8 @@ Thankfully, we have a second dataset that maps the two letter code
 into the full airline name.
 
 ```{code-cell} python
-carrier_code = qeds.load("airline_carrier_codes")
+url = "https://datascience.quantecon.org/assets/data/carrier_code.csv"
```
Comment (wupeifan) [inline lectures/pandas/merge.md:671]: `airline_carrier_codes`

File: lectures/pandas/groupby.md [hunk 5]
```diff
@@ -543,7 +544,8 @@ We'll first look at the data, and then describe the exercise
 random.seed(42)
 np.random.seed(42)
 
-orders = qeds.data.shopify.simulate_orders(500000)
+url = "https://datascience.quantecon.org/assets/data/orders.csv"
```
Comment (wupeifan) [inline lectures/pandas/groupby.md:547]: `shopify_orders.csv`

File: lectures/pandas/merge.md (hunk 3 above)
Comment (Alim-faraji) [reply lectures/pandas/merge.md:640]: I instead changed it to `air_perf = pd.read_csv(url)[["CRSDepTime", "Carrier", "CarrierDelay", "ArrDelay"]]` to do it in one line, like before when using `qeds` package.

**Comment by Alim-faraji** [comment]
I reverted the names to what they were before. Also, some files were more than 50 MB, and Github warned about the file size. So I thought it's better to zip them all.

File: lectures/pandas/merge.md (hunk 3 above)
Comment (wupeifan) [reply lectures/pandas/merge.md]: that's fine

**Comment by wupeifan** [comment]
That's fine; is there anyway that we can check the build results for this PR? Or it's impossible because the data is not copied under gh_pages branch yet?

**Comment by Alim-faraji** [comment]
I think we need to publish first. 
`Note: this requires a publish step for the assets to be copied into gh-pages host`

**Comment by wupeifan** [comment]
Hi @mmcky , I quote Alim's comment on slack here

> But we had `kc_house_data.csv` under a different directory, not in `lectures/_data` . And on that issue, why there are some assets in `lectures/_data` and some others under other directories?

Are there any reasons that we have those assets under different folders? If not, what should we do? Thanks

**Comment by mmcky** [comment]
thanks @wupeifan. @Alim-faraji I am not sure why there are assets in different folders. I would recommend centralising them in a systematic way during this migration.

**Comment by mmcky** [comment]
> That's fine; is there anyway that we can check the build results for this PR? Or it's impossible because the data is not copied under gh_pages branch yet?

we can check this PR via a manual update to `gh-pages` I can do this if you like.

**Comment by wupeifan** [comment]
@Alim-faraji OK then it seems that moving them into the same folder is the right way to go, and check the associated links used in the lectures/problem sets too.

**Comment by Alim-faraji** [comment]
I moved all assets to `lectures/_data`. However, there are some instances of files that are generated and saved in a lecture (`avalanche_incidents.csv` in [here](https://datascience.quantecon.org/applications/working_with_text.html), and `df1.csv` in [here](https://datascience.quantecon.org/pandas/storage_formats.html)). I left them untouched.

**Comment by Alim-faraji** [comment]
Also, [pre_ahs.R](https://github.com/QuantEcon/lecture-datascience.myst/blob/main/lectures/problem_sets/prep_ahs.R) downloads a file, modifies it, and saves three datasets in the same directory. I moved them to `lectures/_data` and replaced the links where they were used.

**Comment by wupeifan** [comment]
> Also, pre_ahs.R downloads a file, modifies it, and saves three datasets in the same directory. I moved them to lectures/_data and replaced the links where they were used.

That R file is never used anywhere, and it prepares the data for one of the problem sets. We already have those files, so you can just move the R file to `lectures/data`.

**Comment by wupeifan** [comment]
> However, there are some instances of files that are generated and saved in a lecture (avalanche_incidents.csv in here)

We need to save this file because in the lecture it fetches from some unstable API.

**Comment by wupeifan** [comment]
After these, @Alim-faraji you can **squash and merge** this one, and let's see what it builds.

**Pull request merged**